Module-12-SpringBoot/src/main/java/com/example/SpringBoot/Module2/SpringBoot/Services/Implementations/UserServiceImplementations.java

```diff
@@ -0,0 +1,140 @@
+        List<User> userList = new ArrayList<User>();
+        userList = (List<User>) userRepository.findAll();
+
+        List<UserDto> returnValue = new ArrayList<UserDto>();
+
+        for (int i = 0; i < userRepository.count(); i++) {
+            returnValue.add(new UserDto());
+        }
+
+        for (int i = 0; i < userRepository.count(); i++) {
+            BeanUtils.copyProperties(userList.get(i), returnValue.get(i));
+        }
+
+        return returnValue;
+    }
+
+    @Override
+    public UserDto findUserByEmail(String emailAddress) {
+        UserDto foundDtoUser = new UserDto();
+        User foundUser = userRepository.findByEmailAddress(emailAddress);
+
+        BeanUtils.copyProperties(foundUser, foundDtoUser);
+
+        return foundDtoUser;
+    }
+
+    @Override
+    public UserDto findUserByUserId(String userId) {
+        UserDto foundDtoUser = new UserDto();
+        User foundUser = userRepository.findByUserId(userId);
+
+        BeanUtils.copyProperties(foundUser, foundDtoUser);
+
+        return foundDtoUser;
+    }
+
+    @Override
+    public UserDto updateUserByEmail(String email, UserDto requestedUpdate) {
+        User updateUser = new User();
+        BeanUtils.copyProperties(requestedUpdate, updateUser);
+
+        User oldUserData = userRepository.findByEmailAddress(email);
+        UserDto returnValue = new UserDto();
+
+        updateUser.setId(oldUserData.getId());
+        updateUser.setUserId(oldUserData.getUserId());
+        updateUser.setEncryptedPassword("test1");
+        updateUser.setEmailVerificationStatus(oldUserData.isEmailVerificationStatus());
+
+        User updatedUserDetails = userRepository.save(updateUser);
+        BeanUtils.copyProperties(updatedUserDetails, returnValue);
+
+        return returnValue;
+    }
+
+    @Override
+    public UserDto updateUserById(String userId, UserDto requestedUpdate) {
+
+        User updateUser = new User();
+        BeanUtils.copyProperties(requestedUpdate, updateUser);
+
+        User oldUserData = userRepository.findByUserId(userId);
+        UserDto returnValue = new UserDto();
+
+        updateUser.setId(oldUserData.getId());
+        updateUser.setUserId(oldUserData.getUserId());
+        updateUser.setEncryptedPassword("test1");
+        updateUser.setEmailVerificationStatus(oldUserData.isEmailVerificationStatus());
+
+        User updatedUserDetails = userRepository.save(updateUser);
+
+        BeanUtils.copyProperties(updatedUserDetails, returnValue);
+
+        return returnValue;
+    }
+
+    @Override
+    public UserDto deleteUserById(String userId) {
+        UserDto foundDtoUser = new UserDto();
+        User foundUser = userRepository.findByUserId(userId);
+
+        BeanUtils.copyProperties(foundUser, foundDtoUser);
+        userRepository.deleteById(foundUser.getId());
+
+        return foundDtoUser;
+    }
+
+    @Override
+    public UserDto deleteUserByEmail(String emailAddress) {
+        UserDto foundDtoUser = new UserDto();
```

Comment:
what if you don't find the user? Throw an exception. This project needs Exception handling